main.tf

```diff
@@ -1,3 +1,65 @@
+# Required for public ECR where Karpenter artifacts are hosted
+provider "aws" {
+  region = "us-west-2"
+  alias  = "oregon"
+}
+
+provider "kubernetes" {
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+  token                  = data.aws_eks_cluster_auth.this.token
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks_blueprints.eks_cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+    token                  = data.aws_eks_cluster_auth.this.token
+  }
+}
+
+provider "kubectl" {
+  apply_retry_count      = 10
+  host                   = module.eks_blueprints.eks_cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+  load_config_file       = false
+  token                  = data.aws_eks_cluster_auth.this.token
+}
+
+module "eks_blueprints" {
+
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.27.0"
+  cluster_name    = local.name
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.vpc.vpc_id
+  private_subnet_ids = module.vpc.private_subnets
+
+  # EKS CONTROL PLANE VARIABLES
+  cluster_version = local.cluster_version
+
+  # List of Additional roles admin in the cluster
+  # Comment this section if you ARE NOT at an AWS Event, as the TeamRole won't exist on your site, or replace with any valid role you want
+  map_roles = [
+    {
+      rolearn  = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/TeamRole"
+      username = "ops-role" # The user name within Kubernetes to map to the IAM role
+      groups   = ["system:masters"] # A list of groups within Kubernetes to which the role is mapped; Checkout K8s Role and Rolebindings
+    }
+  ]
+
+  # EKS MANAGED NODE GROUPS
```

Comment:
Difference b/w Managed and un-managed nodes?

Reply:
With a un-managed node there is a lot you are responsible for configuring. That includes installing the kubelet, container runtime, connecting to the cluster, auto scaling, networking, and more. Whereas managed node groups handle the lifecycle of each worker node for you. A managed node group will come with all the prerequisite software and permissions, connect itself to the cluster, and provide an easier experience for lifecycle actions like auto scaling and updates. So, managed nodes are more easier to use than un-managed nodes.